core/helpers.py

```diff
@@ -11,6 +11,13 @@
 )
 
 
+def instantiate_subclass(base_class, subclass_name):
+    subclasses = {}
+    for cls in base_class.__subclasses__():
+        subclasses[cls.__name__] = cls
+    return subclasses.pop(subclass_name, None)
```

Comment:
I think you want to use `.get` instead of `.pop`, since pop removes the key from the dictionary. In this case we don't care, but it's clearer to use get.

Also, what happens if the user provides a type that doesn't exist? Isn't it better to raise an exception here and catch it in `add_text()`?

Reply:
I handled the case without raising an exception but by returning None. I think it's a cleanest way ?